elastic/apm-server
# aggregation: Add dimensions to transaction metrics

## Motivation/summary

Adds more dimensions that will be added to the aggregated transaction
metrics generated by the APMServer. The added dimensions are:

- `cloud.account.id`
- `cloud.account.name`
- `cloud.machine.type`
- `cloud.project.id`
- `cloud.project.name`
- `cloud.service.name`
- `service.language.name`
- `service.language.version`
- `service.runtime.name`
- `service.runtime.version`
- `host.os.platform`
- `faas.coldstart`
- `faas.trigger.type`

This list is based on https://github.com/elastic/apm/discussions/548.

## Checklist




## How to test these changes

1. `make build-package` 
2. `docker-compose up -d`
3. `make`
4. `./apm-server -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -e`
5. Install the APM integration `8.1.0` (locally built) through Kibana.
6. `curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/transactions.ndjson http://localhost:8200/intake/v2/events`
7. Navigate to Kibana Dev tools and issue the following query
```javascript
GET metrics-apm.internal*/_search
{
  "_source": false,
  "fields": [
    "cloud.account.id",
    "cloud.account.name",
    "cloud.machine.type",
    "cloud.project.id",
    "cloud.project.name",
    "cloud.service.name",
    "service.language.name",
    "service.language.version",
    "service.runtime.name",
    "service.runtime.version",
    "host.os.platform",
    "faas.coldstart",
    "faas.trigger.type"
  ],
  "query": {
    "match": {
      "metricset.name": "transaction"
    }
  }
}
```

Verify that the docs contain all the requested fields. `faas.*` fields must be present in only one of the results.

https://p.elstc.co/paste/yOd+Jknl#FuZqfb-ekPS1pvAdqaI2LprdGdoQcQOyxDh0XA0cPRs



## Related issues

Part of #6843.
